app/api/base_api.rb

```diff
@@ -18,8 +20,11 @@ module BaseAPI
 
     helpers ::Helpers::ErrorHelper
 
-    http_basic do |_username, password|
-      password == Rails.configuration.x.app_password
-    end
+    use Grape::Middleware::OptionalAuth,
+        type: :http_basic,
+        realm: 'API Authorization',
+        proc: ->(_username, password) do
+          password == Rails.configuration.x.app_password
```

Comment:
We can't do that check in the existing block because the middleware checks for the presence of required headers and returns unauthorized before executing the block. 

I could have simply made the addition of the existing middleware conditional on configuration when the application loads, but I prefer the approach taken here because it allows us to test with both configurations.